cli_tools/gce_ovf_import/ovf_importer/ovf_importer.go

```diff
@@ -388,6 +335,54 @@ func (oi *OVFImporter) setUpImportWorkflow() (workflow *daisy.Workflow, err erro
 		return nil, fmt.Errorf("error parsing workflow %q: %v", oi.workflowPath, err)
 	}
 	workflow.ForceCleanupOnError = true
+
+	workflow.SetLogProcessHook(daisyutils.RemovePrivacyLogTag)
+	// See workflows in `ovfWorkflowDir` for variable name declaration.
+	createInstanceStepName := "create-instance"
+	cleanupStepName := "cleanup"
+
+	var dataDiskPrefix string
+	if oi.params.IsInstanceImport() {
+		dataDiskPrefix = oi.params.InstanceNames
+	} else {
+		dataDiskPrefix = oi.params.MachineImageName
+	}
+
+	daisyovfutils.CreateDisksOnInstance(
+		workflow.Steps[createInstanceStepName].CreateInstances.Instances[0],
+		dataDiskPrefix, oi.imageURIs[1:])
+
+	// Delete the images after the instance is created.
+	workflow.Steps[cleanupStepName].DeleteResources.Images = append(
+		workflow.Steps[cleanupStepName].DeleteResources.Images, oi.imageURIs[1:]...)
+	oi.updateImportedInstance(workflow)
+	if oi.params.IsMachineImageImport() {
+		oi.updateMachineImage(workflow)
+	}
+
+	rl := &daisyutils.ResourceLabeler{
+		BuildID:         oi.params.BuildID,
+		UserLabels:      oi.params.UserLabels,
+		BuildIDLabelKey: "gce-ovf-import-build-id",
+		ImageLocation:   oi.imageLocation,
+		InstanceLabelKeyRetriever: func(instanceName string) string {
+			if strings.ToLower(oi.params.InstanceNames) == instanceName {
+				return "gce-ovf-import"
+			}
+			return "gce-ovf-import-tmp"
+		},
+		DiskLabelKeyRetriever: func(disk *daisy.Disk) string {
+			return "gce-ovf-import-tmp"
+		},
+		ImageLabelKeyRetriever: func(imageName string) string {
+			return "gce-ovf-import-tmp"
+		}}
+	rl.LabelResources(workflow)
+	daisyutils.UpdateAllInstanceNoExternalIP(workflow, oi.params.NoExternalIP)
```

Comment:
These 3 steps (remove privacy hook from above, resource labeler and no external IP) are repeated in all operations (except no external IP for export since it doesn't need, but it would be a no-op). They also all operate on the workflow, even though for e.g. bootable disk processor, it's implemented as a processor's method (it still only operates on the workflow). To avoid code duplication, ideally we would have a master function/class that would do these things for all wrappers. This could be done in a followup PR if you prefer.

There's also applying UEFI compatible which doesn't make sense on exports, but it could be theoretically done by the same master function/class, which would skip over UEFI flag for exports.

Reply:
Opened  a bug internally